crypto/src/main/java/org/springframework/security/crypto/factory/RsaKeyConverters.java

```diff
@@ -0,0 +1,123 @@
+ */
+
+package org.springframework.security.crypto.factory;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.util.Assert;
+
+/**
+ * Used for creating {@link java.security.Key} converter instances
+ *
+ * @author Josh Cummings
+ * @since 5.2
+ */
+public class RsaKeyConverters {
+	private static final String DASHES = "-----";
+	private static final String PKCS8_PEM_HEADER = DASHES + "BEGIN PRIVATE KEY" + DASHES;
+	private static final String PKCS8_PEM_FOOTER = DASHES + "END PRIVATE KEY" + DASHES;
+	private static final String X509_PEM_HEADER = DASHES + "BEGIN PUBLIC KEY" + DASHES;
+	private static final String X509_PEM_FOOTER = DASHES + "END PUBLIC KEY" + DASHES;
+
+	/**
+	 * Construct a {@link Converter} for converting a PEM-encoded PKCS#8 RSA Private Key
+	 * into a {@link RSAPrivateKey}.
+	 *
+	 * Note that keys are often formatted in PKCS#1 and this can easily be identified by the header.
+	 * If the key file begins with "-----BEGIN RSA PRIVATE KEY-----", then it is PKCS#1. If it is
+	 * PKCS#8 formatted, then it begins with "-----BEGIN PRIVATE KEY-----".
+	 *
+	 * @return A {@link Converter} that can read a PEM-encoded PKCS#8 RSA Private Key and return a
+	 * {@link RSAPrivateKey}.
+	 */
+	public static Converter<InputStream, RSAPrivateKey> pkcs8() {
+		KeyFactory keyFactory = rsaFactory();
+		return source -> {
+			BufferedReader reader = new BufferedReader(new InputStreamReader(source));
+			List<String> lines = reader.lines().collect(Collectors.toList());
+			Assert.isTrue(!lines.isEmpty() && lines.get(0).startsWith(PKCS8_PEM_HEADER),
+					"Key is not in PEM-encoded PKCS#8 format");
+			String base64Encoded = lines.stream()
+					.filter(RsaKeyConverters::isNotPkcs8Wrapper)
+					.collect(Collectors.joining());
+			byte[] pkcs8 = Base64.getDecoder().decode(base64Encoded);
+
+			try {
+				return (RSAPrivateKey) keyFactory.generatePrivate(
+						new PKCS8EncodedKeySpec(pkcs8));
+			} catch (Exception e) {
+				throw new IllegalArgumentException(e);
+			}
+		};
+	}
+
+	/**
+	 * Construct a {@link Converter} for converting a PEM-encoded X.509 RSA Public Key
+	 * into a {@link RSAPublicKey}.
+	 *
+	 * @return A {@link Converter} that can read a PEM-encoded X.509 RSA Public Key and return a
+	 * {@link RSAPublicKey}.
+	 */
+	public static Converter<InputStream, RSAPublicKey> x509() {
+		KeyFactory keyFactory = rsaFactory();
+		return source -> {
+			BufferedReader reader = new BufferedReader(new InputStreamReader(source));
```

Comment:
Perhaps I'm missing something, but shouldn't we close the `InputStream` after reading it? If not, we should make it clear we don't close the `InputStream` in the Javadoc and explain why.

Reply:
Good catch. My intent was to not make an assumption about the nature of the stream being passed in. So, I'll add a Javadoc to both methods.

That said, I think it makes sense to close it in the post processor since it's the post processor that opens it, so I'll change that, too.